File: pos_bahrain/public/js/scripts/payment_entry.js
async function set_invoice_date(frm, cdt, cdn) {
  console.log('Setter');
  const { reference_doctype, reference_name } = frappe.get_doc(cdt, cdn) || {};
  if (reference_name) {
    const date_field =
      reference_doctype === 'Sales Order' ? 'transaction_date' : 'posting_date';
    const { message: doc = {} } = await frappe.db.get_value(
      reference_doctype,
      reference_name,
      date_field
    );
    return frappe.model.set_value(cdt, cdn, 'pb_invoice_date', doc[date_field]);
  }
}

const payment_entry_reference = {
  refresh: set_invoice_date,
  reference_name: set_invoice_date,
};

export default {
  payment_entry_reference,
};

File: pos_bahrain/doc_events/payment_entry.py
# -*- coding: utf-8 -*-
# Copyright (c) 2019, 	9t9it and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe


def before_save(doc, method):
    for ref in doc.references:
        if not ref.pb_invoice_date:
            date_field = (
                "transaction_date"
                if ref.reference_doctype == "Sales Order"
                else "posting_date"
            )
            ref.pb_invoice_date = frappe.db.get_value(
                ref.reference_doctype, ref.reference_name, date_field
            )
